conans/test/functional/compression/compression_test.py

```diff
@@ -0,0 +1,537 @@
+        with chdir(tmpdir):
+            os.link("a", "b")
+            return os.stat("a").st_ino == os.stat("b").st_ino
+    except OSError:
+        return False
+    finally:
+        shutil.rmtree(tmpdir)
+
+
+@parameterized_class(compressors)
+@attr("slow")
+class CompressionTest(unittest.TestCase):
+    @classmethod
+    def get_file_list(cls, d):
+        files = []
+        dirs = []
+        links = []
+        for root, dirnames, filenames in os.walk(d):
+            for filename in filenames:
+                relpath = os.path.relpath(os.path.join(root, filename), d)
+                if os.path.islink(os.path.join(root, filename)):
+                    links.append(relpath)
+                else:
+                    files.append(relpath)
+            for dirname in dirnames:
+                relpath = os.path.relpath(os.path.join(root, dirname), d)
+                if os.path.islink(os.path.join(root, dirname)):
+                    links.append(relpath)
+                else:
+                    dirs.append(relpath)
+        return files, dirs, links
+
+    @classmethod
+    def is_broken_link(cls, link):
+        if os.path.islink(link):
+            target = os.readlink(link)
+            if not os.path.isabs(target):
+                target = os.path.join(os.path.dirname(link), target)
+            return not os.path.exists(target)
+        return False
+
+    @classmethod
+    def get_inodes(cls, d, files):
+        inodes = dict()
+        for filename in files:
+            stat = os.stat(os.path.join(d, filename))
+            if stat.st_ino not in inodes:
+                inodes[stat.st_ino] = []
+            inodes[stat.st_ino].append(filename)
+        return sorted(list(inodes.values()))
+
+    def assertDirsEqual(self, d1, d2):
+        files1, dirs1, links1 = self.get_file_list(d1)
+        files2, dirs2, links2 = self.get_file_list(d2)
+        self.assertEqual(files1, files2)
+        self.assertEqual(dirs1, dirs2)
+        self.assertEqual(links1, links2)
+
+        for link in links1:
+            filename1 = os.path.join(d1, link)
+            filename2 = os.path.join(d2, link)
+
+            self.assertTrue(os.path.islink(filename2))
+            target1 = os.readlink(filename1)
+            target2 = os.readlink(filename2)
+            self.assertEqual(target1, target2)
+
+            self.assertEqual(self.is_broken_link(filename1), self.is_broken_link(filename2))
+
+        for filename in files1:
+            filename1 = os.path.join(d1, filename)
+            filename2 = os.path.join(d2, filename)
+
+            # don't try to run on symlinks - they might be broken!
+            stat1 = os.stat(filename1)
+            stat2 = os.stat(filename2)
+            self.assertEqual(stat1.st_size, stat2.st_size)
+            # 1. verify contents
+            self.assertEqual(md5sum(filename1), md5sum(filename2))
+            # 2. verify timestamps
+            t1 = int(stat1.st_mtime)
```

Comment:
Is the conversion to int() necessary even if no need to apply the mask because ZipCompressor?

Reply:
let's try to remove it. AFAIK the (one) problem is that python's `stat.st_mtime` is represented as float.

Reply:
removed cast here, it worked